### DISCUSSION TITLE
Double bonds lost in Maximum Common Substructure: Is there a way to get them back?

### DISCUSSION THREAD

**Discussion by jrkarki** [rdkit/rdkit#6031] [Q&A]
Hi everyone,

I performed MCS on s1 and s2 using rdFMCS.FindMCS(mols):

s1 = 'COC1=CC(C2=CN(C)C(=O)C3=CN=CC=C23)=CC(OC)=C1CN1CCN(CCOCCOCC(=O)NCC2=CC=C(S(=O)(=O)NC3=CC=CC4=C3[NH]C=C4Cl)C=C2)CC1'

s2 = 'NS(=O)(=O)C1=CC=C(S(=O)(=O)NC2=CC=CC3=C2[NH]C=C3Cl)C=C1'

The output is:
mcs = 'O=S(=O)(NC1:C:C:C:C2:C1:N:C:C2Cl)C1:C:C:CC:C:1'
mcs_filtered = 'O=S(=O)(NC1CCCC2C(Cl)CNC12)C1CCCCC1'

As you can see on the figure, the double bonds in mcs are lost. Is there a way to resolve this?

Thanks in advance,
Reagon

![Full_mol](https://user-images.githubusercontent.com/25939327/215445466-bedcdbec-bab3-4680-a19d-06ae90dc7300.png)
![s2](https://user-images.githubusercontent.com/25939327/215445680-1123e9d0-acfc-4b98-8c2f-43be1180261f.png)
![mcs1](https://user-images.githubusercontent.com/25939327/215445699-83e2463e-dc7e-4cb3-8e8d-bd8a31b2374b.png)
![mcs2](https://user-images.githubusercontent.com/25939327/215445710-b1fbaf79-62a9-4efb-bb7c-806673707535.png)

**Comment by greglandrum** [comment]
@jrkarki please share the exact code that you used or it's impossible to answer your question

**Reply by jrkarki** [reply]
Here it is @greglandrum:

from rdkit import Chem
from rdkit.Chem import rdFMCS

s1 = 'COC1=CC(C2=CN(C)C(=O)C3=CN=CC=C23)=CC(OC)=C1CN1CCN(CCOCCOCC(=O)NCC2=CC=C(S(=O)(=O)NC3=CC=CC4=C3[NH]C=C4Cl)C=C2)CC1'
s1 = Chem.MolFromSmiles(s1)
s2 = 'NS(=O)(=O)C1=CC=C(S(=O)(=O)NC2=CC=CC3=C2[NH]C=C3Cl)C=C1'
s2 = Chem.MolFromSmiles(s2)

mols = [s1,s2]
mcs = rdFMCS.FindMCS(mols)
mcs = Chem.MolFromSmarts(mcs.smartsString)

mcsTosmiles = Chem.MolToSmiles(mcs)
print(mcsTosmiles)
'O=S(=O)(NC1:C:C:C:C2:C(Cl):C:N:C:1:2)C1:C:C:C:C:C:1'

mcsTosmiles_filtered = mcsTosmiles.replace(":","")
print(mcsTosmiles_filtered)
'O=S(=O)(NC1CCCC2C(Cl)CNC12)C1CCCCC1'
